pkg/syndesis/operation/attach.go

```diff
@@ -0,0 +1,96 @@
+package operation
+
+import (
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"github.com/syndesisio/syndesis-operator/pkg/apis/syndesis/v1alpha1"
+	"github.com/syndesisio/syndesis-operator/pkg/syndesis/template"
+	"github.com/syndesisio/syndesis-operator/pkg/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func AttachSyndesisToResource(syndesis *v1alpha1.Syndesis) error {
+
+	resTypes, err := getAllManagedResourceTypes()
+	if err != nil {
+		return err
+	}
+	selectors := getAllManagerSelectors()
+
+	for _, selector := range selectors {
+		for _, metaType := range resTypes {
+
+			options := sdk.WithListOptions(&selector)
+			list := metav1.List{
+				TypeMeta: metaType,
+			}
+			if err := sdk.List(syndesis.Namespace, &list, options); err != nil {
+				return err
+			}
+
+			for _, obj := range list.Items {
+				res, err := util.LoadKubernetesResource(obj.Raw)
+				if err != nil {
+					return err
+				}
+				SetNamespaceAndOwnerReference(res, syndesis)
+				if err := sdk.Update(res); err != nil {
+					return err
+				}
+			}
+
+		}
+	}
+
+	return nil
+}
+
+func getAllManagerSelectors() []metav1.ListOptions {
+	return []metav1.ListOptions {
+		{
+			LabelSelector: "syndesis.io/app=syndesis,syndesis.io/type=infrastructure",
+		},
+		{
+			LabelSelector: "app=syndesis,syndesis.io/app=todo",
+		},
+		{
+			LabelSelector: "syndesis.io/app=syndesis,syndesis.io/component=syndesis-db",
+		},
+	}
+}
+
+func getAllManagedResourceTypes() ([]metav1.TypeMeta, error) {
+	templTypes, err := template.GetDeclaredResourceTypes()
+	if err != nil {
+		return nil, err
+	}
+	// Using a map to remove duplicates
+	metas := make(map[string]metav1.TypeMeta, 0)
+	for _, kType := range templTypes {
+		addResourceTypes(metas, kType.APIVersion, kType.Kind)
```

Comment:
I think you could use `kType` directly as a key in a map, avoiding to create the key yourself.

AFAIK the only requirement for a key is that it needs to be comparable using `==`. `metav1.TypeMeta` should be comparable as it is a struct with two strings only, and a struct comparison boils down to the comparison of its elements.

Reply:
So you could use a `map[metav1.TypeMeta]bool` as a Set. Actually, you could even change `template.GetDeclaredResourceTypes` to return that set or already a unique slice.

But that should not hinder us to merge that PR now. 'happy to pick this up to find out whether we could use TypeMeta as key.

Reply:
I assumed that complex types as map keys were forbidden in Go.. bu I'll let you investigate :)